### DISCUSSION TITLE
how get httpOnly cookie in loader

### DISCUSSION THREAD

**Discussion by l1mindr** [remix-run/remix#7534] [Q&A] [answered]
Hello, how can I receive the ‍`httpOnly‍` cookies in the loader?

**Comment by sergiodxa** [comment]
All cookies come in the `Cookie` header, so you can do `request.headers.get("cookie")` to get that value, then parse it with something, if you used Remix's Cookie object (created with createCookie) you can use `cookie.parse` to get the value of that specific cookie, if the cookie was created another way you can parse it with `cookie-js` or any other library.

**Reply by l1mindr** [reply]
I wanted to send the cookies with fetch, but it seems that the cookies are not sent
```
const res = await fetch('url', {
    credentials: 'include',
  })
```

**Reply by sergiodxa** [reply, marked as answer]
That only works client-side. Server-side `fetch` has no way to know the cookies the request received because it's not tied to the request in any way.

What you need to do is to grab the Cookie header and add it to the fetch yourself.

```ts
await fetch(url, {
  headers: { cookie: request.headers.get("cookie") }
})
```

You could even pass the whole headers object if you don't mind sending every header to the URL you're fetching.

```ts
await fetch(url, { headers: request.headers })
```

**Reply by l1mindr** [reply]
ty
